lib/global-styles.php

```diff
@@ -291,90 +291,117 @@ function gutenberg_experimental_global_styles_get_core() {
 /**
  * Returns the theme presets registered via add_theme_support, if any.
  *
+ * @param array $settings Existing editor settings.
+ *
  * @return array Config that adheres to the theme.json schema.
  */
-function gutenberg_experimental_global_styles_get_theme_support_settings() {
+function gutenberg_experimental_global_styles_get_theme_support_settings( $settings ) {
 	$theme_settings                       = array();
 	$theme_settings['global']             = array();
 	$theme_settings['global']['settings'] = array();
 
 	// Deprecated theme supports.
-	if ( get_theme_support( 'disable-custom-colors' ) ) {
+	if (
+		array_key_exists( 'disableCustomColors', $settings ) &&
+		$settings['disableCustomColors']
+	) {
 		if ( ! isset( $theme_settings['global']['settings']['color'] ) ) {
 			$theme_settings['global']['settings']['color'] = array();
 		}
 		$theme_settings['global']['settings']['color']['custom'] = false;
+		unset( $settings['disableCustomColors'] );
```

Comment:
So the issue is, that unlike in the previous implementation, here settings are unset only when they are truthy.

Reply:
That was one of the issues, others were the order in which our filter ran, but also not unsetting them at the proper place. This is now fixed.